src/video/SDL_egl.c

```diff
@@ -1278,12 +1281,86 @@ SDL_EGL_DestroySurface(_THIS, EGLSurface egl_surface)
     if (!_this->egl_data) {
         return;
     }
-    
+
     if (egl_surface != EGL_NO_SURFACE) {
         _this->egl_data->eglDestroySurface(_this->egl_data->egl_display, egl_surface);
     }
 }
 
+SDL_GLImageKHR SDL_EGL_CreateImageDmabuf(_THIS, const SDL_ImageDmabuf *buffer)
+{
+    EGLint eglImgAttrs[30];
+    unsigned int atti = 0;
+    EGLImageKHR image;
+    uint32_t format;
+
+    if (!buffer || !buffer->num_plane)
+        return NULL;
+
+    switch (buffer->format) {
+    case SDL_PIXELFORMAT_NV12:
+        format = DRM_FORMAT_NV12;
+        break;
+    case SDL_PIXELFORMAT_IYUV:
+        format = DRM_FORMAT_YUV420;
+        break;
+    case SDL_PIXELFORMAT_YUY2:
+        format = DRM_FORMAT_YUYV;
+        break;
+    default:
+        return NULL;
+    }
+
+    eglImgAttrs[atti++] = EGL_WIDTH;
+    eglImgAttrs[atti++] = buffer->width;
+    eglImgAttrs[atti++] = EGL_HEIGHT;
+    eglImgAttrs[atti++] = buffer->height;
+    eglImgAttrs[atti++] = EGL_LINUX_DRM_FOURCC_EXT;
+    eglImgAttrs[atti++] = format;
+
+    if (buffer->num_plane > 0 && buffer->planes[0].fd >= 0) {
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE0_FD_EXT;
+        eglImgAttrs[atti++] = buffer->planes[0].fd;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE0_OFFSET_EXT;
+        eglImgAttrs[atti++] = buffer->planes[0].offset;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE0_PITCH_EXT;
+        eglImgAttrs[atti++] = buffer->planes[0].stride;
+    }
+
+    if (buffer->num_plane > 1 && buffer->planes[1].fd >= 0) {
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE1_FD_EXT;
+        eglImgAttrs[atti++] = buffer->planes[1].fd;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE1_OFFSET_EXT;
+        eglImgAttrs[atti++] = buffer->planes[1].offset;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE1_PITCH_EXT;
+        eglImgAttrs[atti++] = buffer->planes[1].stride;
+    }
+
+    if (buffer->num_plane > 2 && buffer->planes[2].fd >= 0) {
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE2_FD_EXT;
+        eglImgAttrs[atti++] = buffer->planes[2].fd;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE2_OFFSET_EXT;
+        eglImgAttrs[atti++] = buffer->planes[2].offset;
+        eglImgAttrs[atti++] = EGL_DMA_BUF_PLANE2_PITCH_EXT;
+        eglImgAttrs[atti++] = buffer->planes[2].stride;
+    }
+
+    eglImgAttrs[atti++] = EGL_NONE;
+
+    image = _this->egl_data->eglCreateImageKHR(_this->egl_data->egl_display,
```

Comment:
`eglCreateImageKHR` and `eglDestroyImageKHR` are extension functions that may not be present. You must check that they are not NULL before calling them.

Reply:
yes, you're right